src/lib.rs

```diff
@@ -927,12 +938,16 @@ fn fill_todo(
             let dirs = fs::read_dir(path).and_then(|d| {
                 d.map(|e| {
                     e.map(|e| {
-                        let path = if curdir {
-                            PathBuf::from(e.path().file_name().unwrap())
+                        let (path, file_name) = if curdir {
+                            let path = e.path();
+                            let file_name = path.file_name().unwrap();
+                            (PathBuf::from(file_name), file_name.to_owned())
                         } else {
-                            e.path()
+                            let path = e.path();
+                            let file_name = path.file_name().unwrap().to_owned();
```

Comment:
Before, we did not have the `unwrap` here. Could you please remove it and pass `Option<OsString>` to `from_dir_entry` instead? Thank you!